app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java

```diff
@@ -234,11 +235,25 @@ public Task<Integer> calcPinCost(@NonNull Location location) {
                 .continueWith(task -> (Integer) task.getResult().getData());
     }
 
+    public Task<Pin> findPin(String pid, Location location) {
+        HashMap<String, Object> data = new HashMap<>();
+        data.put("pid", pid);
+        data.put("latitude", location.getLatitude());
+        data.put("longitude", location.getLongitude());
+
+        return functions.getHttpsCallable("findPin").call(data).continueWith(task -> {
```

Comment:
Do we want to try catch for errors from the cloud function? Otherwise it'll crash the app if we e.g. pass a self-owned pin to this function.

Reply:
Will it actually crash the app? I think if the cloud function throws an error it will get caught in the `onFailureListener` in the `MapFragment` which displays the error message to the user.

Reply:
Yep you're right, I was thinking we used an `onCompleteListener` which has a habit of crashing the app.